book/features/external_toc.md

```diff
@@ -0,0 +1,69 @@
+
+## What does it do?
+
+This functionality, part of the [TeachBooks package](https://github.com/TeachBooks/TeachBooks) allows you to refer to `.ipynb` and `.md` files stored on public repositories. These files are then handles as if they were normal files in your book, leading to a the file as a page in your built book. To prevent issues it warns you if there's a mismatch in `requirements.txt`, combines `.bib`-files, merges plugins in `_config.yml` and checks licenses.
+
+## Installation
+To use this extenstion, follow these steps:
+
+**Step 1: Install the Package**
+
+Install the module `teachbooks` package using `pip`:
+```
+pip install git+https://github.com/BSchilperoort/TeachBooks@external-git-content
+```
+    
+**Step 2: Add to `requirements.txt`**
+
+Make sure that the package is included in your project's `requirements.txt` to track the dependency:
+```
+git+https://github.com/BSchilperoort/TeachBooks@external-git-content
+```
+
+## Usage
+
+To use the functionality, take the URL of a file and add it to your `_toc.yml` as follows:
+
+```yaml
+format: jb-book
+root: .
+
+parts:
+  - caption: ...
+    chapters:
+    - file: ...
+    .
+    .
+    .
+    - external: <link to GitHub / GitLab source .md or .ipynb file>
+    .
+    .
+    .
+```
+
+The link can be pointing to a file on a branch (which will take the most recent version of that file on the branch) or a commit or tag (which will take a specific version of the file). For example:
+- `external: https://github.com/TeachBooks/manual/blob/release/book/intro.md` will take the most recent version of the intro page of this book on the release branch
+- `external: https://github.com/TeachBooks/manual/blob/06d67e8a0110c94d9147ce07090656dedc7d0e64/README.md` will take the file during the state of a specific commit (06d67e8).
+- `external: https://github.com/TeachBooks/manual/blob/v1.1.1/README.md` will take the file during the state of a specific tag (v1.1.1)
+
+If you're building your book online in GitHub, the [deploy-book-workflow](../external/deploy-book-workflow/README.md) will deal with these files during the `teachbooks build` command. If you want to build the book locally, run the command `teachbooks build`.
+
+If the source file is from another book, the contents of the `requirements.txt`, `_config.yml` and `*.bib` file are merged to make sure the page has the same functionality as in the source repository. If this leads to compatibility issue, you'll be warned during the build. Furthermore, licenses of external books will be checked to avoid copyright issues.
```

Comment:
```suggestion
If the source file is from another book, the contents of the `requirements.txt`, `_config.yml` are checked for any missing/incompatible entries. If this leads to compatibility issue, you'll be warned during the build which will help you solve these dependencies or plugins manually by updating your main book's `_config.yml` and `requirements.txt`
Additionally, any `references.bib` files are merged into your main book's `references.bib` file, and the licenses of the external content are validated to allow for re-use. External content without a permissive license will result in an error to try to prevent any accidental copyright infringement.
```